docs/examples/soiling/plot_greensboro_kimber_soiling.py

```diff
@@ -30,17 +30,16 @@
 # step.
 
 from datetime import datetime
-import pathlib
 from matplotlib import pyplot as plt
 from pvlib.iotools import read_tmy3
 from pvlib.soiling import kimber
-import pvlib
+from pvlib.tools import dataset
 
 # get full path to the data directory
-DATA_DIR = pathlib.Path(pvlib.__file__).parent / 'data'
+DATASET_DIR = dataset('723170TYA.CSV')
```

Comment:
Looking at this example code, I think we should consider more descriptive names.  I'm worried that people will see it in an example and think that it's for retrieving all datasets (whatever that means), not just the example datasets included in `pvlib.data.`  `locate_example_dataset()`, maybe?

I'd also name this variable `tmy_filepath` or something like that, and update the comment since it's now a specific file instead of a directory.

Reply:
Fair point, although in that sense I think (English is not my mother tongue) something along `get_dataset_path` makes more sense, maybe? Just to bring up other options, I'll stick to your suggestion for now.

Thanks for spotting that of the comments and the var name.

Reply:
Now I am wondering if having `locate` in the function name might make people think it has something to do with `pvlib.location.Location`...

I like that `get_dataset_path` makes it clear that the function returns a path to a dataset (not the dataset itself), but I still think the name should also make it clear that it is just for example/test datasets.  `get_test_dataset_path`?

Reply:
That's a good point. I like `get_test_dataset_path`.
If I or anyone addresses #1056, another function called `get_example_dataset_path` would also make sense. Do you think that this one should be the latter, since the current path is considered to remain for example files (from [this comment](https://github.com/pvlib/pvlib-python/issues/1056#issuecomment-689714408))? And leave the former for test files?

I will rename it tomorrow.

Reply:
Good question.  On first thought, I don't think we should have a function for each folder, since depending on where we go with #1056, we might end up with multiple `data` folders inside `tests`.  What do you think about having a single function, but with an optional `location='data'` parameter to allow finding the file in a custom folder?  For example `get_test_dataset_path(dataset='723170TYA.CSV', location='tests/data')`?  And then the path would look like `dataset = pathlib.Path(pvlib.__path__[0], location, dataset)`.

Reply:
I couldn't like that approach more.
Just a quick detail, should `location` default to user example data? Or do we prefer being verbose on that?

Reply:
Yeah, it makes sense to me for it to default to `data` so files will be found in `pvlib/data` by default.  That means for now the `location` parameter probably won't be useful to anyone, since all files are in `pvlib/data` right now. But if we expand to have multiple `data` folders in the future, then it will be useful.

I guess that means maybe we shouldn't bother adding the parameter yet, since there is no use for it today?  We could leave that for future work, and just leave a note about it in #1056.